static/app/views/replays/detail/network/index.tsx

```diff
@@ -0,0 +1,185 @@
+import {Fragment, useCallback, useMemo, useState} from 'react';
+import styled from '@emotion/styled';
+
+import {PanelTable, PanelTableHeader} from 'sentry/components/panels';
+import Placeholder from 'sentry/components/placeholder';
+import {showPlayerTime} from 'sentry/components/replays/utils';
+import {IconArrow} from 'sentry/icons';
+import {t} from 'sentry/locale';
+import space from 'sentry/styles/space';
+import {EventTransaction} from 'sentry/types';
+import theme from 'sentry/utils/theme';
+
+import {ISortConfig, NetworkSpan, sortNetwork} from './utils';
+
+type Props = {
+  event: EventTransaction;
+  networkSpans: NetworkSpan[];
+};
+
+function NetworkList({event, networkSpans}: Props) {
+  const {startTimestamp} = event;
+  const [sortConfig, setSortConfig] = useState<ISortConfig>({
+    by: 'startTimestamp',
+    asc: true,
+  });
+
+  const handleSort = useCallback(
+    (
+      heading: typeof sortConfig.by,
+      substractValue?: typeof sortConfig.substractValue
+    ) => {
+      setSortConfig(prevSort => {
+        if (prevSort.by === heading) {
+          return {by: heading, asc: !prevSort.asc, substractValue};
+        }
+
+        return {by: heading, asc: true, substractValue};
+      });
+    },
+    [sortConfig]
+  );
+
```

Comment:
I think the interface for this method could be more generalized by having the callsite pass in a method instead of specifically asking for a 2nd field name.

For example, what if the interface was something like:
```
handleSort(fieldName: keyof NetworkSpan);
handleSort(key: string, getValue: (row: NetworkSpan) => any = null);
handleSort(fieldName: keyof NetworkSpan, getValue: (row: NetworkSpan) => any = null) {
  if (getValue === null) {
    getValue = (row) => row[fieldName];
  }
  ... snip ...
}
```

In this proposal the first 2 lines are for typescript, they tell TS that there are two signatures allowed:
a) either pass in a valid `keyof NetworkSpan` and we will sort with that
B) OR pass in a string (which is our key for sorting asc/desc) and also a getter method. We can pass in a getter method that does the work to return duration.